src/domain/Phone/WebRTCPhone.js

```diff
@@ -184,14 +184,36 @@ export default class WebRTCPhone extends Emitter implements Phone {
     return true;
   }
 
-  sendReinvite(sipSession: Session, newConstraints: Object = null) {
-    logger.info('WebRTC phone - send reinvite', { sessionId: sipSession ? sipSession.id : null, newConstraints });
+  async sendReinvite(sipSession: Session, newConstraints: Object = null, conference: boolean = false) {
+    let constraints = newConstraints;
+    logger.info('WebRTC phone - send reinvite', { sessionId: sipSession ? sipSession.id : null, constraints });
 
     if (!sipSession) {
       return;
     }
 
-    return this.client.reinvite(sipSession, newConstraints);
+    const callSession = this.callSessions[this.getSipSessionId(sipSession)];
+    const hasRemoteVideo = callSession ? !!this.getRemoteVideoReceiver(callSession) : false;
+    // Do not recreate a stream in 1:1 when a remote video stream is already present. Just replace the video track.
+    if (callSession && constraints && constraints.video && hasRemoteVideo && !conference) {
+      // Reset session video device id to create new stream
+      await this.changeSessionVideoInputDevice(null);
+      // $FlowFixMe
+      await this.client.setVideoInputDevice(typeof newConstraints.video === 'string' ? newConstraints.video
```

Comment:
What kind of string value can `newConstraints.video` take? Shouldn't it be a boolean or an object?

Reply:
For the record: The boolean/object format is only for the low-level (WebRTC) constraints. The higher-level Wazo SDK accepts constraints as boolean or string, where the string is the device ID.